cirq-core/cirq/value/classical_data.py

```diff
@@ -0,0 +1,278 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import enum
+from typing import Dict, Mapping, Sequence, Tuple, TYPE_CHECKING, TypeVar
+
+from cirq.value import digits, value_equality_attr
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class MeasurementType(enum.IntEnum):
```

Comment:
Do you know how this interacts with [previous issues with enum serialization](https://github.com/quantumlib/Cirq/issues/3853) that we've seen?

Reply:
IntEnum serializes to an integer (and deserializes too). It took a while to figure out a way to get it to all work, but this seems to do everything we need it to. I may have missed something, but I *think* this is a valid approach. LMK if you see anything I missed.